domain/src/main/kotlin/no/nav/su/se/bakover/domain/beregning/Beregning.kt

```diff
@@ -49,3 +52,27 @@ data class Beregning(
         }
     }
 }
+
+private fun fradragWithForventetInntekt(fradrag: List<Fradrag>, forventetInntekt: Int): MutableList<Fradrag> {
+    val newFradrag: MutableList<Fradrag> = mutableListOf()
+
+    if (fradrag.isEmpty()) {
+        newFradrag.add(Fradrag(type = Fradragstype.ForventetInntekt, beløp = forventetInntekt))
+    } else {
+        val arbeidsInntektTotalt = fradrag.filter { it.type == Fradragstype.Arbeidsinntekt }.sumBy { it.beløp }
+
+        if (forventetInntekt > arbeidsInntektTotalt) {
+            newFradrag.add(Fradrag(type = Fradragstype.ForventetInntekt, beløp = forventetInntekt))
+        }
+
+        for (f in fradrag) {
+            if (f.type == Fradragstype.Arbeidsinntekt && arbeidsInntektTotalt >= forventetInntekt) {
+                newFradrag.add(f)
+            } else if (f.type != Fradragstype.Arbeidsinntekt) {
+                newFradrag.add(f)
+            }
+        }
+    }
+
+    return newFradrag
```

Comment:
Jeg syns denne var litt tricky å henge med på, så jeg gjorde et forsøk på å forenkle den. Såvidt jeg forstod så skal dette gjøre det samme.

```suggestion
val (arbeidsinntektfradrag, andrefradrag) = fradrag.partition { it.type === Fradragstype.Arbeidsinntekt }
val totalArbeidsinntekt = arbeidsinntektfradrag.sumBy { it.beløp }

if (totalArbeidsinntekt > forventetInntekt) {
    return fradrag
}
return andrefradrag.plus(Fradrag(type = Fradragstype.ForventetInntekt, beløp = forventetInntekt))
```

Reply:
ja, var akkurat det jeg prøvde på, uten at jeg visste at partition og plus var en greie. 🥇 🥇